dds/DCPS/transport/rtps_udp/RtpsUdpDataLink.cpp

```diff
@@ -2948,6 +2917,78 @@ OpenDDS::DCPS::RtpsUdpDataLink::receive_strategy()
   return static_cast<OpenDDS::DCPS::RtpsUdpReceiveStrategy*>(receive_strategy_.in());
 }
 
+OPENDDS_SET(ACE_INET_Addr)
+OpenDDS::DCPS::RtpsUdpDataLink::get_addresses(const RepoId& local, const RepoId& remote) const {
+  OPENDDS_SET(ACE_INET_Addr) retval;
+
+  get_addresses_i(local, remote, retval);
+
+  return retval;
+}
+
+OPENDDS_SET(ACE_INET_Addr)
+OpenDDS::DCPS::RtpsUdpDataLink::get_addresses(const RepoId& local) const {
+  OPENDDS_SET(ACE_INET_Addr) retval;
+
+  const GUIDSeq_var peers = peer_ids(local);
+  if (peers.ptr()) {
+    for (CORBA::ULong i = 0; i < peers->length(); ++i) {
+      get_addresses_i(local, peers[i], retval);
+    }
+  }
+
+  return retval;
+}
+
+void
+OpenDDS::DCPS::RtpsUdpDataLink::get_addresses_i(const RepoId& local, const RepoId& remote,
```

Comment:
Should this be called `accumulate_addresses`?

Reply:
Done.